libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp

```diff
@@ -20,19 +20,38 @@ VoteManager::VoteManager(addr_t node_addr, std::shared_ptr<DbStorage> db, std::s
   LOG_OBJECTS_CREATE("VOTE_MGR");
 
   // Retrieve votes from DB
-  daemon_ = std::make_unique<std::thread>([this]() { retreieveVotes_(); });
+  daemon_ = std::make_unique<std::thread>([this]() {
+    stopped_ = false;
+    retreieveVotes_();
+  });
 
   current_period_final_chain_block_hash_ = final_chain_->block_header()->hash;
 }
 
-VoteManager::~VoteManager() { daemon_->join(); }
+VoteManager::~VoteManager() {
+  stopped_ = true;
+  daemon_->join();
+}
 
 void VoteManager::setNetwork(std::weak_ptr<Network> network) { network_ = std::move(network); }
 
 void VoteManager::retreieveVotes_() {
   LOG(log_si_) << "Retrieve verified votes from DB";
   auto verified_votes = db_->getVerifiedVotes();
   const auto pbft_step = db_->getPbftMgrField(PbftMgrRoundStep::PbftStep);
+  for (auto const& v : verified_votes) {
+    addVerifiedVote(v);
+    LOG(log_dg_) << "Retrieved verified vote " << *v;
+  }
+
+  // TODO: Implement votes sync properly
+  // Since this is invoked on startup no peers are connected yet, invoke two minutes after startup
+  for (int i = 0; i < 120; i++) {
+    if (stopped_ == false) {
```

Comment:
Since retreieveVotes_ only be called once in constructor, I think we don't need the "stopped_" variable. Why we need the stopped_ here?